docs/troubleshooting.asciidoc

```diff
@@ -90,3 +90,21 @@ For this you'll need an `IApmLogger` implementation (see above) which you need t
 ----
 Agent.Setup(new AgentComponents(logger: new ApmLoggerBridge()));
 ----
+
+[float]
+[[double-agent-initialization]]
+=== An `InstanceAlreadyCreatedException` exception is thrown
+
+It can happen that in a very early stage of the monitored process the agent throws an `InstanceAlreadyCreatedException` exception with the following message: "The singleton APM agent has already been instantiated and can no longer be configured." This happens when the agent would be initialized multiple times, which is prohibited.
+
+NOTE: The reason for forcing a single agent instance per process is that enabling multiple agents would open up problems like capturing events and metrics multiple times for each agent instance, or having multiple background threads for event serialization and transfer to the APM server.
+
+Take a look at the initialization section of the <<public-api, Public Agent API>> for more information on how agent initialization works.
+
+As an example, this issue can happen if you call the `Elastic.Apm.Agent.Setup` method multiple times, or if you call another method on `Elastic.Apm.Agent` that implicitly initializes the agent and then you call the `Elastic.Apm.Agent.Setup` method on the already initialized agent.
+
+Another scenario can be that you use the Public Agent API in combination of the IIS module or the ASP.NET Core NuGet package where you enable the agent with the `UseElasticApm` or `UseAllElasticApm` methods. Both the first call to the IIS module and the `UseElasticApm`/`UseAllElasticApm` methods internally call the `Elastic.Apm.Agent.Setup` method to initialize the agent.
+
+In case you use the Public Agent API with the `Elastic.Apm.Agent` class that can be potentially executed before the IIS module initializes or the `UseElasticApm`/`UseAllElasticApm` calls execute, then those will fail, since you implicitly initialized the agent before.
+
+To prevent the `InstanceAlreadyCreatedException` in those scenarios, in case you have code that can potentially run very early before the agent is initialized, it's recommended to check with the `Elastic.Apm.Agent.IsConfigured` method whether the agent is already initialized and only then use other methods in the Public Agent API.
```

Comment:
I'm still struggling with this troubleshooting section. I attempted a rewrite. Feel free to take some, all, or none:

```
In the early stage of a monitored process, the Agent might throw an `InstanceAlreadyCreatedException` exception with the following message: "The singleton APM agent has already been instantiated and can no longer be configured." This happens when you attempt to initialize the Agent multiple times, which is prohibited. Allowing multiple Agent instances per process would open up problems, like capturing events and metrics multiple times for each instance, or having multiple background threads for event serialization and transfer to the APM Server.

TIP: Take a look at the initialization section of the <<public-api,Public Agent API>> for more information on how agent initialization works.

As an example, this issue can happen if you call the `Elastic.Apm.Agent.Setup` method multiple times, or if you call another method on `Elastic.Apm.Agent` that implicitly initializes the agent, and then you call the `Elastic.Apm.Agent.Setup` method on the already initialized agent.

Another example might be when you use the Public Agent API in combination with the IIS module or the ASP.NET Core NuGet package, where you enable the agent with the `UseElasticApm` or `UseAllElasticApm` methods. Both the first call to the IIS module and the `UseElasticApm`/`UseAllElasticApm` methods internally call the `Elastic.Apm.Agent.Setup` method to initialize the agent.

If you use the Public Agent API with the `Elastic.Apm.Agent` class, there's a chance it can execute before the IIS module initializes or the `UseElasticApm`/`UseAllElasticApm` calls execute. If that happens, those will fail, as the Agent has been implicitly initialized already.

To prevent the `InstanceAlreadyCreatedException` in these scenarios, first use the `Elastic.Apm.Agent.IsConfigured` method to check if the agent is already initialized. After the check, you can safely use other methods in the Public Agent API. This will prevent code from running before the Agent has initialized.
```

Reply:
Nice, thanks! 

I took most of it, changed 2 things:

> If you use the Public Agent API with the Elastic.Apm.Agent class, there's a chance it can execute before the IIS module initializes or the UseElasticApm/UseAllElasticApm calls execute. If that happens, those will fail, as the Agent has been implicitly initialized already.

I think the "there's a chance it can execute before..." part is a bit unfortunate so I changed that part. Both the IIS module and the `UseElasticApm/UseAllElasticApm` methods run super early, at a very well defined stage in the application startup. I wanted the text to reflect that this issue happens when users use the public API extremely early in the application process. So this does not happen "by chance" this needs active work and thinking by the user to find a point to run code before the agent initializes :)

(last sentence)
> This will prevent code from running before the Agent has initialized.

Changed to:

> This will prevent accidental implicit agent initialization.